Formula/haproxy.rb

```diff
@@ -13,6 +13,9 @@ class Haproxy < Formula
 
   depends_on "openssl"
   depends_on "pcre"
+  depends_on "lua" => :optional
+
+  patch :DATA
```

Comment:
What's going on here?

Reply:
@DomT4 a simple patch at end of this formula to remove `--export-dynamic` from `ld` options because it isn't supported by macOS ld.

It doesn't requested before because this part of Makefile is enabled only by `USE_LUA=1`.

I've [send this patch also to upstream](https://www.mail-archive.com/haproxy@formilux.org/msg28408.html) but anyway it wasn't backported to 1.8 branch I guess.

Reply:
Has upstream accepted it on another branch, or is it still pending review?

Reply:
@DomT4 it was send at same time when I created this pull request. So, I think that it is too early to be accepted to master branch.

Reply:
It's likely worthwhile waiting on an upstream response here.

Reply:
@DomT4 np, I guess they're responding in few days because it is a holiday season :)

Reply:
Trying to get that line in the file to behave properly cross-platform is a worthwhile goal, but I don't think we actually have to patch. Doesn't
```
      args << "LUA_LD_FLAGS=-L#{lua.lib}"
```
take care of it without further intervention?

Reply:
`opt_lib` and `opt_include` would make me less twitchy about growing inconsistencies 🙈.

Reply:
you need to click refresh

Reply:
❤️

Reply:
@ilovezfs honestly I think this two line was a side effect of `export-dynamic`. Because I added them before the patch.

Let me try to cock the formula without this hack.

Reply:
Yes, it builds and works without it.

Shall I make a force push or add one more commit?

Reply:
force push please :)

Reply:
@ilovezfs done!

Reply:
hmm I'm still seeing the patch

Reply:
@ilovezfs I mean remove `LUA_LIB` and `LUA_INC` :( I can't remove the patch because ld doesn't support `export-dynamic` at macOS that is hardcoded at Makefile.

Reply:
it will override it.

Reply:
@ilovezfs I think I don't understand you.

Without this patch the formula doesn't build with lua with errors like:
```
Updating Homebrew...
==> Downloading https://www.haproxy.org/download/1.8/src/haproxy-1.8.2.tar.gz
Already downloaded: /Users/catap/Library/Caches/Homebrew/haproxy-1.8.2.tar.gz
==> make CC=clang CFLAGS= LDFLAGS= TARGET=generic USE_KQUEUE=1 USE_POLL=1 USE_PCRE=1 USE_OPENSSL=1 USE_ZLIB=1 ADDLIB=-lcrypto USE_LUA=1
Last 15 lines from /Users/catap/Library/Logs/Homebrew/haproxy/01.make:

make
CC=clang
CFLAGS=
LDFLAGS=
TARGET=generic
USE_KQUEUE=1
USE_POLL=1
USE_PCRE=1
USE_OPENSSL=1
USE_ZLIB=1
ADDLIB=-lcrypto
USE_LUA=1

Makefile:638: *** unable to automatically detect the Lua library name, you can enforce its name with LUA_LIB_NAME=<name> (where <name> can be lua5.3, lua53, lua, ...).  Stop.
```

If I added `LUA_LIB_NAME` or `LUA_LIB` and `LUA_INC` it doesn't build with error likes:
```
clang -Iinclude -Iebtree -Wall        -DTPROXY -DUSE_ZLIB  -DENABLE_POLL -DENABLE_KQUEUE -DUSE_OPENSSL  -DUSE_LUA  -DUSE_PCRE -I/usr/local/Cellar/pcre/8.41/include  -DCONFIG_HAPROXY_VERSION=\"1.8.2-08396fa\" -DCONFIG_HAPROXY_DATE=\"2017/12/23\" -c -o src/base64.o src/base64.c
clang -Iinclude -Iebtree -Wall        -DTPROXY -DUSE_ZLIB  -DENABLE_POLL -DENABLE_KQUEUE -DUSE_OPENSSL  -DUSE_LUA  -DUSE_PCRE -I/usr/local/Cellar/pcre/8.41/include  -DCONFIG_HAPROXY_VERSION=\"1.8.2-08396fa\" -DCONFIG_HAPROXY_DATE=\"2017/12/23\" -c -o src/hash.o src/hash.c
clang  -o haproxy src/ev_poll.o src/ev_kqueue.o src/ssl_sock.o src/hlua.o src/hlua_fcn.o ebtree/ebtree.o ebtree/eb32sctree.o ebtree/eb32tree.o ebtree/eb64tree.o ebtree/ebmbtree.o ebtree/ebsttree.o ebtree/ebimtree.o ebtree/ebistree.o src/proto_http.o src/cfgparse.o src/server.o src/stream.o src/flt_spoe.o src/stick_table.o src/stats.o src/mux_h2.o src/checks.o src/haproxy.o src/log.o src/dns.o src/peers.o src/standard.o src/sample.o src/cli.o src/stream_interface.o src/proto_tcp.o src/backend.o src/proxy.o src/tcp_rules.o src/listener.o src/flt_http_comp.o src/pattern.o src/cache.o src/filters.o src/vars.o src/acl.o src/payload.o src/connection.o src/raw_sock.o src/proto_uxst.o src/flt_trace.o src/session.o src/ev_select.o src/channel.o src/task.o src/queue.o src/applet.o src/map.o src/frontend.o src/freq_ctr.o src/lb_fwlc.o src/mux_pt.o src/auth.o src/fd.o src/hpack-dec.o src/memory.o src/lb_fwrr.o src/lb_chash.o src/lb_fas.o src/hathreads.o src/chunk.o src/lb_map.o src/xxhash.o src/regex.o src/shctx.o src/buffer.o src/action.o src/h1.o src/compression.o src/pipe.o src/namespace.o src/sha1.o src/hpack-tbl.o src/hpack-enc.o src/uri_auth.o src/time.o src/proto_udp.o src/arg.o src/signal.o src/protocol.o src/lru.o src/hdr_idx.o src/hpack-huff.o src/mailers.o src/h2.o src/base64.o src/hash.o    -lz  -lssl -lcrypto -Wl,--export-dynamic  -llua -lm -L/usr/local/Cellar/pcre/8.41/lib -lpcreposix -lpcre -lcrypto
ld: unknown option: --export-dynamic
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make: *** [haproxy] Error 1
```

I can't overwrite `-Wl,--export-dynamic` because it is [hardoced at Makefile](http://git.haproxy.org/?p=haproxy.git;a=blob;f=Makefile;h=2acf502811c0008f1593edb03413057576c82b7e;hb=HEAD#l633).

So, I have only one way include a trivial patch that removes `-Wl,--export-dynamic`.

Reply:
```
    if build.with?("lua")
      lua = Formula["lua"]
      args << "USE_LUA=1"
      args << "LUA_LIB=#{lua.opt_lib}"
      args << "LUA_INC=#{lua.opt_include}"
      args << "LUA_LD_FLAGS=-L#{lua.opt_lib}"
    end
```

Reply:
@ilovezfs ah, it is `:=`

Ok, you're right.